libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -152,6 +152,54 @@ string YulUtilFunctions::storeLiteralInMemoryFunction(string const& _literal)
 	});
 }
 
+string YulUtilFunctions::copyLiteralToStorageFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_storage_" + util::toHex(util::keccak256(_literal).asBytes());
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers templ(R"(
+			function <functionName>(slot) {
+				<body>
+			}
+		)");
+		templ("functionName", functionName);
```

Comment:
Yeah, the split version is much nicer indeed now. I wonder whether we should have some helper for doing this boilerplate here, though. Maybe even a toplevel member of ``m_functionCollector``, s.t. this would read
```
m_functionCollector.createFunctionWithArgsAndBody(functionName, [&](auto& args, auto& body) {
  args = "slot";
  body = Whiskers(...)(...).render();
});
```
(Or with another similar mechanism, the named output arguments were just a quick and dirty idea right now.)
But: different PR for sure.

Reply:
I think that makes a lot of sense, as many (if not all of calls to `createFunction`) start with:
```
string functionName = "...";
return m_functionCollector.createFunction(functionName, [&]() {
	Whiskers templ(R"(
		function <functionName>(<args>) -> <returns> {
```

It would simplify not only places where we have `<body>` template :)

Reply:
I'll open issue for this now.

Reply:
I did add a quick and first draft version of yul function collector update. It is here #11066.
Once that, or something similar gets merged, we could use that interface to update this one